### ISSUE TITLE
NullPointerException: null

### ISSUE THREAD

**Issue by ghost** [Vektah/CodeGlance#64]
It has happened only once so far, and it seemed to occur doing a quick switch between windows.

No debug details where provided (AFAIS) except the error message in subject title. I know this is not much to go by, but I am logging the error here for the record. I'll add more details if/when they should be available.

Version IDE: WebStorm 9.0.2 (build WS-139.773)
Plugin: 1.4.2
Windows: 8.1 x64

**Comment by epistemex** [comment]
More details available:

`null
java.lang.NullPointerException
	at net.vektah.codeglance.render.Minimap.getLine(Minimap.java:137)
	at net.vektah.codeglance.render.CoordinateHelper.offsetToScreenSpace(CoordinateHelper.java:146)
	at net.vektah.codeglance.GlancePanel.paintSelection(GlancePanel.java:279)
	at net.vektah.codeglance.GlancePanel.paint(GlancePanel.java:234)
	at javax.swing.JComponent.paintToOffscreen(JComponent.java:5219)
	at javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(RepaintManager.java:1529)
	at javax.swing.RepaintManager$PaintManager.paint(RepaintManager.java:1452)
	at javax.swing.RepaintManager.paint(RepaintManager.java:1249)
	at javax.swing.JComponent._paintImmediately(JComponent.java:5167)
	at javax.swing.JComponent.paintImmediately(JComponent.java:4978)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:808)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:796)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:796)
	at javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:769)
	at javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:718)
	at javax.swing.RepaintManager.access$1100(RepaintManager.java:62)
	at javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1677)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:312)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:733)
	at java.awt.EventQueue.access$200(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:694)
	at java.awt.EventQueue$3.run(EventQueue.java:692)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:703)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:748)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:577)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)`

**Comment by TonyCode** [comment]
Same Exception, but slightly different stacktrace:

* IDE: IntelliJ IDEA 14.0.3
* CodeGlance plugin's version: 1.4.2
* Win 7, x64

```
null
java.lang.NullPointerException
	at net.vektah.codeglance.render.Minimap.getLine(Minimap.java:137)
	at net.vektah.codeglance.render.CoordinateHelper.offsetToCharacterInLine(CoordinateHelper.java:158)
	at net.vektah.codeglance.GlancePanel.paintSelection(GlancePanel.java:278)
	at net.vektah.codeglance.GlancePanel.paint(GlancePanel.java:234)
	at javax.swing.JComponent.paintToOffscreen(JComponent.java:5223)
        ...
```

**Comment by ksdev-pl** [comment]
Hello! 

Another NullPointerException:

```
null
java.lang.NullPointerException
	at net.vektah.codeglance.render.Minimap.getLine(Minimap.java:137)
	at net.vektah.codeglance.render.CoordinateHelper.offsetToScreenSpace(CoordinateHelper.java:146)
	at net.vektah.codeglance.GlancePanel.paintSelection(GlancePanel.java:277)
	at net.vektah.codeglance.GlancePanel.paint(GlancePanel.java:234)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JLayeredPane.paint(JLayeredPane.java:585)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at com.intellij.ui.tabs.impl.JBTabsImpl.paintChildren(JBTabsImpl.java:2291)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at com.intellij.ui.tabs.impl.JBTabsImpl.paint(JBTabsImpl.java:2286)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JLayeredPane.paint(JLayeredPane.java:585)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at com.intellij.openapi.wm.impl.ToolWindowsPane.paintChildren(ToolWindowsPane.java:185)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JLayeredPane.paint(JLayeredPane.java:585)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paint(JComponent.java:1064)
	at javax.swing.JLayeredPane.paint(JLayeredPane.java:585)
	at javax.swing.JComponent.paintChildren(JComponent.java:888)
	at javax.swing.JComponent.paintToOffscreen(JComponent.java:5239)
	at javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(RepaintManager.java:1529)
	at javax.swing.RepaintManager$PaintManager.paint(RepaintManager.java:1452)
	at javax.swing.RepaintManager.paint(RepaintManager.java:1249)
	at javax.swing.JComponent.paint(JComponent.java:1041)
	at java.awt.GraphicsCallback$PaintCallback.run(GraphicsCallback.java:39)
	at sun.awt.SunGraphicsCallback.runOneComponent(SunGraphicsCallback.java:78)
	at sun.awt.SunGraphicsCallback.runComponents(SunGraphicsCallback.java:115)
	at java.awt.Container.paint(Container.java:1967)
	at java.awt.Window.paint(Window.java:3877)
	at com.intellij.openapi.wm.impl.IdeFrameImpl.paint(IdeFrameImpl.java:513)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:819)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:796)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:796)
	at javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:769)
	at javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:718)
	at javax.swing.RepaintManager.access$1100(RepaintManager.java:62)
	at javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1677)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:312)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:738)
	at java.awt.EventQueue.access$300(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:699)
	at java.awt.EventQueue$3.run(EventQueue.java:697)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:708)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:748)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:577)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)

```

**Comment by nazrhyn** [comment]
I just got this same exception. Here's another stack trace for you:
```
null
java.lang.NullPointerException
	at net.vektah.codeglance.render.Minimap.getLine(Minimap.java:137)
	at net.vektah.codeglance.render.CoordinateHelper.offsetToCharacterInLine(CoordinateHelper.java:158)
	at net.vektah.codeglance.GlancePanel.paintSelection(GlancePanel.java:278)
	at net.vektah.codeglance.GlancePanel.paint(GlancePanel.java:234)
	at javax.swing.JComponent.paintToOffscreen(JComponent.java:5219)
	at javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(RepaintManager.java:1529)
	at javax.swing.RepaintManager$PaintManager.paint(RepaintManager.java:1452)
	at javax.swing.RepaintManager.paint(RepaintManager.java:1249)
	at javax.swing.JComponent._paintImmediately(JComponent.java:5167)
	at javax.swing.JComponent.paintImmediately(JComponent.java:4978)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:808)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:796)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:796)
	at javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:769)
	at javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:718)
	at javax.swing.RepaintManager.access$1100(RepaintManager.java:62)
	at javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1677)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:312)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:733)
	at java.awt.EventQueue.access$200(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:694)
	at java.awt.EventQueue$3.run(EventQueue.java:692)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:703)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:734)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:569)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
```

**Comment by nazrhyn** [comment]
:+1: Happened again. v1.4.2 on WebStorm 10.04

**Comment by Vektah** [comment]
fixed in  1.4.3
